services/supervisord/logs.go

```diff
@@ -269,3 +289,157 @@ func readURL(ctx context.Context, url string) ([]byte, error) {
+	if err != nil {
+		return err
+	}
+	if err := tmpfile.Close(); err != nil {
+		return err
+	}
+
+	adminArgs := [][]string{
+		{"summary", "--filename", tmpfile.Name(), "--skip-server"},
+		{"summary", "--filename", tmpfile.Name()},
+	}
+
+	// Try with and without --skip-server just in case the installed pmm-admin version doesn't support it
+	for _, args := range adminArgs {
+		cmd := exec.CommandContext(ctx, "pmm-admin", args...) // nolint
+		pdeathsig.Set(cmd, unix.SIGKILL)
+		var out []byte
+		out, err = cmd.CombinedOutput()
+		log.Println(string(out))
+
+		if err != nil {
+			continue
+		}
+		break
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "cannot run pmm-admin summary")
+	}
+
+	tmpDir, err := ioutil.TempDir("", "pmm-admin-summary")
+	if err != nil {
+		return err
+	}
+
+	if err := unzip(tmpfile.Name(), tmpDir); err != nil {
+		return err
+	}
+
+	clientDir := filepath.Join(tmpDir, "client")
+	if _, err := os.Stat(clientDir); err != nil {
+		return err
+	}
+
+	if err := addToZip(clientDir, archive); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func unzip(archive, target string) error {
+	reader, err := zip.OpenReader(archive)
+	if err != nil {
+		return err
+	}
+
+	if err := os.MkdirAll(target, os.ModePerm); err != nil {
+		return err
+	}
+
+	for _, file := range reader.File {
+		path := filepath.Join(target, filepath.Clean(file.Name))
+		dir := filepath.Dir(path)
+		if _, err := os.Stat(dir); os.IsNotExist(err) {
+			if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+				return err
+			}
+		}
+
+		if file.FileInfo().IsDir() {
+			if err := os.MkdirAll(path, file.Mode()); err != nil {
+				return err
+			}
+			continue
+		}
+
+		fileReader, err := file.Open()
+		if err != nil {
+			return err
+		}
+
+		targetFile, err := os.OpenFile(path, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, file.Mode())
+		if err != nil {
+			return err
+		}
+
+		if _, err := io.Copy(targetFile, fileReader); err != nil {
+			return err
+		}
+
+		fileReader.Close() //nolint:errcheck
+		targetFile.Close() //nolint:errcheck
+	}
+
+	return nil
+}
+
+func addToZip(source string, archive *zip.Writer) error {
+	info, err := os.Stat(source)
+	if err != nil {
+		return nil
+	}
+
+	var baseDir string
+	if info.IsDir() {
+		baseDir = filepath.Base(source)
+	}
+
+	err = filepath.Walk(source, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		header, err := zip.FileInfoHeader(info)
+		if err != nil {
+			return err
+		}
+
+		if baseDir != "" {
+			header.Name = filepath.Join(baseDir, strings.TrimPrefix(path, source))
+		}
+
+		if info.IsDir() {
+			header.Name += "/"
+		} else {
+			header.Method = zip.Deflate
+		}
+
+		writer, err := archive.CreateHeader(header)
+		if err != nil {
+			return err
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		// filepath.Clean is no really needed here since path comes from Walk.
+		// Added to make linters happy without using // nolint
+		file, err := os.Open(filepath.Clean(path))
```

Comment:
There is nothing wrong with adding `//nolint:gosec` – it is visible, can be audited, can be removed as linter improves, etc. This code, on the other hand, is not immediately visible and can't be analysed by a tool.